woocommerce/woocommerce-ios
# Dynamic Dashboard: Add new view for customizing dashboard layout

Closes: #12451 


## Description

This PR adds a new view for customizing dashboard layout. Most of the code was shamelessly copied from `AnalyticsHubCustomizeView` and view model since the layout is similar.

Persistence will be handled in a separate PR, so the Save button in the new view only updates the UI for now.

## Testing instructions

- Log in to a store eligible for Blaze.
- Tap the Settings button on the top right of the Dashboard screen.
- Confirm that the Stats and Blaze options are available on the Customize Dashboard screen.
- Enable/disable or drag to reorder the cards, confirm that the Save button is enabled.
- Tap Save and confirm that the layout of the dashboard screen is updated accordingly.
- Switch to a store ineligible for Blaze.
- Tap the Settings button on the top right of the Dashboard screen.
- Confirm that the Blaze option is displayed at the end of the options on the Customize Dashboard screen, with an `Unavailable` badge.

## Screenshots

Ineligible for Blaze | Eligible for Blaze
--- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/56215227-1220-48da-ad42-046a675552d1" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/09467154-40b4-4f3a-94f7-4dd8c1b4dae0" width=320 />


---
